### DISCUSSION TITLE
Tip! Sipeed RV-Debugger Plus inexpensive open source USB to JTAG/UART debug board

### DISCUSSION THREAD

**Discussion by Hedda** [zigpy/zigpy#762] [General]
"RV Debugger Plus" by Sipeed is a new inexpensive JTAG and UART debug board with USB-C port as open source hardware:

https://twitter.com/SipeedIO/status/1397203784368107520

https://twitter.com/SipeedIO/status/1406813367860224004

https://www.cnx-software.com/2021/06/21/rv-debugger-plus-uart-jtag-debug-board-bl702-zigbee-ble-risc-v-soc/

It just cost $3.50 at Aliexpress now and should soon also be sold by Seeed Studio as well:

https://www.aliexpress.com/item/1005002837417966.html

RV-Debugger-BL702 open source project which "implements a JTAG + UART debugger firmware inspired by open-ec":

https://github.com/sipeed/RV-Debugger-BL702

![image](https://user-images.githubusercontent.com/6320001/122767409-eb59d500-d2a2-11eb-8971-f79e953f3feb.png)

This is for JTAG/UART debugging but based on Bouffalo Lab BL702 (BL702C-A0) Zigbee & BLE RISC-V MCU SoC.

So this adapter can be used as USB-JTAG debug probe alternative to SEGGER J-Link with example OpenOCD. 

The firmware emulates an FT2232D using the BL702 to implement the device interface of the part of FTDI MPSSE (Multi-Protocol Synchronous Serial Engine) and that can then be used as an JTAG+UART debugger by OpenOCD.

https://github.com/sipeed/RV-Debugger-BL702/tree/main/tools/openocd

http://openocd.org

While could be used as a Zigbee device if solder a ceramic chip antenna, I only posted about it here as UART & JTAG debug board as a cheap adapter tool to flash firmware or program Zigbee dongles/devices or ESP8266/ESP32.

Note that it does not look to support SWD(?).

![image](https://user-images.githubusercontent.com/6320001/122767549-147a6580-d2a3-11eb-86d9-61f617185ba9.png)

![image](https://user-images.githubusercontent.com/6320001/122767590-1d6b3700-d2a3-11eb-966f-d85ccb833b15.png)

https://www.cnx-software.com/2020/10/24/bl602-bl604-risc-v-wifi-bluetooth-5-0-le-soc-will-sell-at-esp8266-price-point/

https://www.bouffalolab.com/bl70X

https://github.com/bouffalolab/bl_mcu_sdk/tree/master/docs/chipSpecification/bl70x_docs

https://github.com/bouffalolab/bl_iot_sdk

PS: Sipeed also makes and sell a ST-LINK V2 STM8/STM32 compatible USB-JTAG/TTL RISC-V debugger as well but it cost much more:

https://www.aliexpress.com/item/1005002714665888.html

Disclaimer: This is brand new hardware from Sipeed that I only just read about so not a product that tested or recommend myself.

**Comment by MattWestb** [comment]
Its looks having full J-Tag connector with all needed pins and SWD is using minimum 2 and more of the j-tag pins and the rest is in software side so i think it shall working with EFR32MG1 chips and perhaps on EFR32MG2 is supported software have its supported newer CPUs.
